internal/agentsession/rollout_path.go

```
@@ -0,0 +1,35 @@
+	root := filepath.Join(home, ".codex", "sessions")
+	rootAbs, err := filepath.Abs(root)
+	if err != nil {
+		return "", err
+	}
+	pathAbs, err := filepath.Abs(path)
+	if err != nil {
+		return "", err
+	}
+	relative, err := filepath.Rel(rootAbs, pathAbs)
+	if err != nil || relative == ".." || strings.HasPrefix(relative, ".."+string(filepath.Separator)) {
+		return "", errors.New("Codex rollout path escapes the sessions root")
```

golangci-lint (2.12.2)

[error] ST1005: error strings should not be capitalized

(staticcheck)